lib/fhir_models/bootstrap/model.rb

```diff
@@ -300,6 +300,26 @@ def check_binding_uri(uri, value)
       valid
     end
 
+    def each_element(path = nil, &block)
```

Comment:
Thoughts on whether this should be `#each_element!` with a bang?

This would be more Ruby idomatic - i.e. [Ruby docs on method syntax](https://ruby-doc.org/core-2.7.2/doc/syntax/methods_rdoc.html) says:

>The bang methods (! at the end of the method name) are called and executed just like any other method. However, by convention, a method with an exclamation point or bang is considered dangerous. In Ruby’s core library the dangerous method implies that when a method ends with a bang (!), it indicates that unlike its non-bang equivalent, permanently modifies its receiver.

@ssayer and I talked about this when using `walk_resources` in https://github.com/dvci/health_cards/pull/60#discussion_r646925122 (which duplicates the bundle to avoid modifying the original)

Reply:
I think it definitely should not be a bang method. It doesn't modify anything.

Reply:
After doing some more testing with this and `#each` I agree.

I was thinking of a use case like

```ruby
# `obs` is a FHIR observation resource I grabbed from the README example
obs.each_element {|v,m,p| p == 'code' ? v.text = 'foo' : v }
```

This does modify the object being acted on. `#each` also does the same though for something like

```ruby
z.each {|v| v[:a] = 2}
z = [{a:1}, {b:2}]

# puts z shows:
# [{:a=>2}, {:b=>2, :a=>2}]
```

I think the nuance I missed here is that `#each`/`#each_element` do not modify the object themselvse, but the block passed in has the potential too (which is kinda a duh moment when I think about it).